contractsV5/KyberNetwork.sol

```diff
@@ -100,12 +98,14 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
     /// @dev add or deletes a reserve to/from the network.
     /// @param reserve The reserve address.
     function addReserve(address reserve, uint reserveId, bool isFeePaying, address wallet) public onlyOperator returns(bool) {
-        require(reserveIdToAddresses[reserveId].length == 0);
-        require(reserveAddressToId[reserve] == uint(0));
+        require(reserveIdToAddresses[reserveId].length == 0, "reserveId points to existing reserve");
+        require(reserveAddressToId[reserve] == uint(0), "reserve has an existing id");
         
         reserveAddressToId[reserve] = reserveId;
 
+        reserveIdToAddresses[reserveId] = new address[](1);
```

Comment:
notice
this array is holding all reserves. also previous ones.
So it will get larger as reserves are added...

Reply:
Fixed